src/services/api/src/queue/coordinator.py

```diff
@@ -125,9 +125,15 @@ def loop(self):
                     self.initialize()
 
                 # Step each processor to advance its state machine.
-                for processor in self.processors.values():
+                for processor in list(self.processors.values()):
                     # TODO catch exceptions and raise them only after all processors are done
-                    processor.step()
+                    try:
+                        processor.step()
+                    except LookupError as e:
```

Comment:
If the `processor.step()` logic raises an exception that isn't a `ValueError`, it still raises. So this should at the very least log the exception if it isn't a `LookupError`, and probably stream something generic to the user.